addons/pos_mercado_pago/static/src/app/payment_mercado_pago.js

```diff
@@ -0,0 +1,219 @@
+                },
+            };
+            // mp_payment_intent_create will call the Mercado Pago api
+            this.env.services.orm.silent.call(
+                "pos.payment.method",
+                "mp_payment_intent_create",
+                [[line.payment_method.id], infos]
+            ).then(result => {
+                resolve(result);
+            });
+        });
+        this.last_status_payment_intent = () => new Promise(resolve => {
+            const line = this.pos.get_order().selected_paymentline;
+            // mp_payment_intent_get will return the last payment intend
+            // recorded in the db by the last authenticated webhook
+            this.env.services.orm.silent.call(
+                "pos.payment.method",
+                "mp_payment_intent_get",
+                [[line.payment_method.id], this.payment_intent.id]
+            ).then(result => {
+                resolve(result);
+            });
+        });
+        this.cancel_payment_intent = () => new Promise(resolve => {
+            const line = this.pos.get_order().selected_paymentline;
+            // mp_payment_intent_cancel will call the Mercado Pago api
+            this.env.services.orm.silent.call(
+                "pos.payment.method",
+                "mp_payment_intent_cancel",
+                [[line.payment_method.id], this.payment_intent.id]
+            ).then(result => {
+                resolve(result);
+            });
+        });
+    }
+
+    async send_payment_request(cid) {
+        await super.send_payment_request(...arguments);
+        const line = this.pos.get_order().selected_paymentline;
+        const MAX_RETRY = 5; // Maximum number of retries for the "ON_TERMINAL" BUG
+        const RETRY_DELAY = 1000; // Delay between retries in milliseconds for the "ON_TERMINAL" BUG
+        try {
+            // During payment creation, user can't cancel the payment intent
+            line.set_payment_status("waitingCapture");
+
+            // Call Mercado Pago to create a payment intent
+            const payment_intent = await this.create_payment_intent();
+
+            // In case of error, don't update 'this.payment_intent' and handle the error
+            if (!("id" in payment_intent)) {
+                switch(payment_intent.status) {
+                    case 409:
+                        // We don't return on this error case, we continue but
+                        // without updating 'this.payment_intent'
+                        this._showMsg("Payment already sent to terminal, check it...", 'information');
+                        break;
+                    case 400: // Payment must be >= 500 [units]
+                    case 401: // Invalid token 
+                    default:
+                        this._showMsg(payment_intent.message, 'error');
+                        return false;
+                  }
+            } else {
+                this.payment_intent = payment_intent
```

Comment:
https://www.digitalocean.com/community/tutorials/linting-and-formatting-with-eslint-in-vs-code